engines/io_uring.c

```diff
@@ -343,6 +365,14 @@ static enum fio_q_status fio_ioring_queue(struct thread_data *td,
 
 	/* ensure sqe stores are ordered with tail update */
 	write_barrier();
+	if (rand_between(&ld->prio_state, 0, 99) < o->hiprio_percent) {
```

Comment:
Probably gate this on o->hiprio_percent being valid. If not set, we shouldn't do anything. This is a hot path, so generating random values if not enabled is a really bad idea. I'd make it:

```
if (o->hiprio_percent && rand_between(&ld->prio_state, 0, 99) < o->hiprio_percent) {
        ...
```

Interestingly, you seem to do this correctly for libaio, but not here.